src/dir.cr

```diff
@@ -246,9 +214,7 @@ class Dir
   # Creates a new directory at the given path. The linux-style permission mode
   # can be specified, with a default of 777 (0o777).
   def self.mkdir(path, mode = 0o777)
-    if LibC.mkdir(path.check_no_null_byte, mode) == -1
-      raise Errno.new("Unable to create directory '#{path}'")
-    end
+    Crystal::System::DirHandle.mkdir(path, mode)
     0
```

Comment:
Not exactly related to this pr, but why does ˋmkdirˋ & ˋrmdirˋ returns ˋ0ˋ?

Reply:
I don't know, but in spec there is
`Dir.mkdir(path, 0o700).should eq(0)`

Reply:
Probably they should be marked as `: Nil` for avoid leaking a useless return value

Reply:
I'll make another PR for it